itest/lib/appStep/api/search.ts

```diff
@@ -21,58 +21,17 @@ export const getResults = async (
   app: Application,
   includeHeaders = true
 ): Promise<string[]> => {
-  // Return search current search results. This is useful for getting
-  // the current results by doing a search through other means, like a
-  // context menu-built search.
-
-  const searchDisplayHeaders = async (app: Application): Promise<string[]> => {
-    // This stinks. We have to use getHTML because headers that are off the
-    // screen return as empty strings if you use getText. This isn't required of
-    // actual results.
-    // See http://v4.webdriver.io/api/property/getText.html
-    // app.browserWindow.maximize() fixes the problem on my laptop but not CI.
-    // But what we get back includes the width which can be non-deterministic:
-    // <div class="header-cell" style="width: 192px;">ts<div class="col-resizer"></div></div>
-    // That style width will vary on my laptop vs. CI.
-    // The hack is to split this and extract just the text.
-    const _trim = (s: string) => s.split(">")[1].split("<")[0]
-
-    let headers = await logStep("get search headers", async () => {
-      const headers = await app.client.$$(selectors.viewer.headers)
-      return await Promise.all(
-        headers.map(async (h) => {
-          await h.waitForDisplayed()
-          return await h.getHTML()
-        })
-      )
-    })
-
-    return headers.map((h) => {
-      return _trim(h)
-    })
-  }
-
-  const searchResults = await logStep("get search records", async () => {
-    const results = await app.client.$$(selectors.viewer.results)
-    return await Promise.all(
-      results.map(async (r) => {
-        await r.waitForDisplayed()
-        return await r.getText()
-      })
-    )
-  })
-
-  let headers
-  if (
-    includeHeaders &&
-    (await (await app.client.$(selectors.viewer.headers)).isDisplayed())
-  ) {
-    headers = await searchDisplayHeaders(app)
-  } else {
-    headers = []
+  const viewer = await app.client.$(".viewer")
+  const html = await viewer.getHTML()
+  document.body.innerHTML = html
```

Comment:
Yeah, good question. I doubt this is how spectron intended us to test the app. But I was thinking about how the "ECONNRESET" errors were happening all over the place and wondering if we were making too many calls for the new web driver server to handle. It's just a hunch, it could be something else. But either way, all the tests will speed up significantly with this pattern. 

Rather than make a request every time we need to look/parse a DOM element, I figured we could get all the DOM we need from webdriver _once_. Then since Jest comes with JSDOM built in, we just load it into the fake Jest DOM and inspect it synchronously without having to talk to another process.